megamek/src/megamek/common/weapons/ArtilleryWeaponIndirectFireHandler.java

```diff
@@ -103,7 +103,16 @@ public boolean handle(GamePhase phase, Vector<Report> vPhaseReport) {
         }
 
         final Vector<Integer> spottersBefore = aaa.getSpotterIds();
-        Coords targetPos = target.getPosition();
+
+        // Handle counter-battery on fleeing/fled off-board targets.
+        Coords targetPos;
+        try {
+            targetPos = target.getPosition();
```

Comment:
If you know what the problem is (target == null), then you may as well check for that instead of using a try/catch block.

Reply:
Yeah, I'd tried a more involved solution before paring it back to this - but had forgotten how much worse the try/catch performance is compared to just `if (null == X)`.